### PR TITLE
Data Loading RFC

### PR DESCRIPTION
**Pull request by ibgreen**
For #

#### Change List
-

### CODE REVIEW

File: dev-docs/RFCs/v7.x/data-loading-rfc.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+```js
+import {parse, registerLoaders} from '@loaders.gl/core';
+import {CSVLoader} from '@loaders.gl/CSV';
+registerLoaders(CSVLoader);
+
+new AnyLayer({
+  coordinateSystem: COORDINATE_SYSTEM.IDENTITY,
+  data: CSV_URL,
+  // NEW: Accept a parse methods
+  parse,
```
Comment (Pessimistress) [inline dev-docs/RFCs/v7.x/data-loading-rfc.md]: How does this work with the current `dataTransform` prop?

**Comment by Pessimistress** [comment]
Shall we add the generator proposal into this RFC?

### Support async generator function provided to `data` Prop

If a function is provided to `data` prop, it is interpreted as a [generator function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*). The [Generator object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator) returned by this function yields batches of data rows or promises that resolve to data rows. Internally, a layer would iterate through the batches, incrementally update the layer whenever a new batch arrives, until all data is loaded.

```js
function loadChunk(index) {
  return d3.csv(`http://data-url/${index}.csv`);
}

async function* getData() {
  for (let i = 0; i < 10; i++) {
    yield await loadChunk(i);
  }
}

new ScatterplotLayer({
  data: getData,
  ...
})
```

File: dev-docs/RFCs/v7.x/data-loading-rfc.md (hunk 1 above)
Comment (ibgreen) [reply dev-docs/RFCs/v7.x/data-loading-rfc.md]: > How does this work with the current dataTransform prop?

`dataTransform` is still applied to the output of `parse`. 

I will clarify in docs.

**Comment by ibgreen** [comment]
> Shall we add the generator proposal into this RFC?

We could, though my intention was to create small "bite sized" RFCs that can be quickly taking through review to get approved (or rejected).

My sense is that there could be a "long tail" of discussion on the streaming RFC on the batch formats, concatenation strategies etc. So I think a separate small data-streaming-rfc.md might make best sense.

I can start it based on your example.

File: dev-docs/RFCs/v7.x/data-loading-rfc.md (hunk 1 above)
Comment (Pessimistress) [reply dev-docs/RFCs/v7.x/data-loading-rfc.md]: Arguably you can already achieve this by supplying your own `fetch` prop. What is the input to the `parse` callback? Do we really need separate props for all these stages? Should they be named in a way that imply they are in the same group?

**Comment by ibgreen** [comment]
I plan to make another update on the details of loaders.gl integration and backwards compatibility. Landing this now and we can review the final changes separately.

**Pull request merged**